server/src/main/java/io/druid/guice/http/HttpClientModule.java

```diff
@@ -135,14 +138,23 @@ public HttpClient get()
     {
       final DruidHttpClientConfig config = getConfigProvider().get().get();
 
+      if (config.getUnusedConnectionTimeout() != null &&
+          config.getUnusedConnectionTimeout().isLongerThan(config.getReadTimeout())) {
```

Comment:
nit: I would use duration.compareTo(otherDuration) >= 0 to incorporate equal value too

Reply:
Ahh.. fixed :)